### PR TITLE
Ui2 first commit

### PR DESCRIPTION
**Pull request by micowan**

### CODE REVIEW

**Review by dthaler — DISMISSED** [review]

File: ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/2HydrophonesController.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+﻿namespace OrcaHello.Web.Api.Controllers
+{
+    [Route("api/hydrophones")]
+    [SwaggerTag("This controller is responsible retrieving Hydrophone locations.")]
```
Comment (dthaler) [inline ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/2HydrophonesController.cs:4]:
```suggestion
    [SwaggerTag("This controller is responsible for retrieving Hydrophone locations.")]
```

File: ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/3DetectionsController.cs [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        }
+
+        [HttpGet]
+        [SwaggerOperation(Summary = "Gets a list of Detections for the passed filter conditions.")]
+        [SwaggerResponse(StatusCodes.Status200OK, "Returns the list of detections for the given filter conditions.", typeof(DetectionListResponse))]
+        [SwaggerResponse(StatusCodes.Status400BadRequest, "If the request was malformed (missing parameters).")]
+        [SwaggerResponse(StatusCodes.Status500InternalServerError, "If there is an internal error reading or processing data from the data source.")]
+        [AllowAnonymous]
+        public async ValueTask<ActionResult<DetectionListResponse>> GetPaginatedDetectionsAsync(
+            [SwaggerParameter("The desired state.", Required = true)] string state,
```
Comment (dthaler) [inline ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/3DetectionsController.cs:120]:
```suggestion
            [SwaggerParameter("The desired state: Unreviewed, Negative, Positive, or Unknown.", Required = true)] string state,
```

File: ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/3DetectionsController.cs [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            [SwaggerParameter("The start date of the search (MM/DD/YYYY).", Required = true)] DateTime? fromDate,
+            [SwaggerParameter("The end date of the search (MM/DD/YYYY).", Required = true)] DateTime? toDate,
+            [SwaggerParameter("The name of the field to sort by (give examples here).", Required = true)] string sortBy,
```
Comment (dthaler) [inline ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/3DetectionsController.cs:123]:
```suggestion
            [SwaggerParameter("The name of the field to sort by (timestamp, locationName, etc.).", Required = true)] string sortBy,
```

File: ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/4CommentsController.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        private readonly ICommentOrchestrationService _commentOrchestrationService;
+
+        public CommentsController(ICommentOrchestrationService commentOrchestrationService)
+        {
+            _commentOrchestrationService = commentOrchestrationService;
+        }
+
+        [HttpGet("negative-unknown")]
+        [SwaggerOperation(Summary = "Gets a paginated list of comments for negative and unknown Detections for the given timeframe.")]
+        [SwaggerResponse(StatusCodes.Status200OK, "Returns the list of comments for the give timeframe.", typeof(CommentListResponse))]
```
Comment (dthaler) [inline ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/4CommentsController.cs:17]:
```suggestion
        [SwaggerResponse(StatusCodes.Status200OK, "Returns the list of comments for the given timeframe.", typeof(CommentListResponse))]
```

File: ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/7TagsController.cs [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+                    exception is TagOrchestrationServiceException)
+                    return Problem(exception.Message);
+
+                return Problem(exception.Message);
+            }
+        }
+
+        [HttpGet("bytimeframe")]
+        [SwaggerOperation(Summary = "Gets a list of tags for the given timeframe.")]
+        [SwaggerResponse(StatusCodes.Status200OK, "Returns the list of tags for the give timeframe.", typeof(TagListForTimeframeResponse))]
```
Comment (dthaler) [inline ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/7TagsController.cs:45]:
```suggestion
        [SwaggerResponse(StatusCodes.Status200OK, "Returns the list of tags for the given timeframe.", typeof(TagListForTimeframeResponse))]
```

File: ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Shared/Models/Hydrophones/Hydrophone.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    [SwaggerSchema("Hydrophone location information.")]
+    public class Hydrophone
+    {
+        [SwaggerSchema("The hydrophone's id.")]
+        public string Id { get; set; }
+
+        [SwaggerSchema("The hydrophone's location name.")]
+        public string Name { get; set; }
+
+        [SwaggerSchema("The hydrophone's location logitude.")]
```
Comment (dthaler) [inline ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Shared/Models/Hydrophones/Hydrophone.cs:13]:
```suggestion
        [SwaggerSchema("The hydrophone's location longitude.")]
```

**Review by dthaler — APPROVED** [review]

**Review by micowan — COMMENTED** [review]
Brought this in under an updated PR

File: ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/2HydrophonesController.cs (hunk 1 above)
Comment (micowan) [reply ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/2HydrophonesController.cs]: Did as part of updated PR

Comment (micowan) [reply ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/2HydrophonesController.cs]: Did as part of updated PR

File: ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/3DetectionsController.cs (hunk 2 above)
Comment (micowan) [reply ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/3DetectionsController.cs]: Did as part of updated PR

File: ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/3DetectionsController.cs (hunk 3 above)
Comment (micowan) [reply ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/3DetectionsController.cs]: Did as part of updated PR

File: ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/4CommentsController.cs (hunk 4 above)
Comment (micowan) [reply ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/4CommentsController.cs]: Did as part of updated PR

File: ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/7TagsController.cs (hunk 5 above)
Comment (micowan) [reply ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Api/Controllers/7TagsController.cs]: Did as part of updated PR

File: ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Shared/Models/Hydrophones/Hydrophone.cs (hunk 6 above)
Comment (micowan) [reply ModeratorFrontEnd/OrcaHello/OrcaHello.Web.Shared/Models/Hydrophones/Hydrophone.cs]: Did as part of updated PR

**Pull request merged**